asn1/InterledgerTypes.asn

```diff
@@ -0,0 +1,165 @@
+InterledgerTypes
+DEFINITIONS
+AUTOMATIC TAGS ::=
+BEGIN
+
+IMPORTS
+    Int16,
+    UInt8,
+    UInt64,
+    VarUInt,
+    Float64
+    FROM GenericTypes
+;
+
+-- Readable names for special characters that may appear in ILP addresses
+hyphen IA5String ::= "-"
+period IA5String ::= "."
+underscore IA5String ::= "_"
+tilde IA5String ::= "~"
+
+-- A standard interledger address
+Address ::= IA5String
+    (FROM
+        ( hyphen
+        | period
+        | "0".."9"
+        | "A".."Z"
+        | underscore
+        | "a".."z"
+        | tilde )
+    )
+    (SIZE (1..1023))
+
+-- --------------------------------------------------------------------------
+
+-- Amounts are represented as IEEE 754 double-precision floating point.
```

Comment:
This section needs to be replaced with an explanation of the choice of UInt64's and the fact that those represent the minimum divisible unit on the destination ledger